src/components/02_atoms/FileUpload/FileUpload.js

```diff
@@ -67,34 +160,187 @@ class FileUpload extends React.Component {
+        }),
+        resetState,
+      );
+      return false;
+    }
+
+    return true;
+    /* eslint-enable camelcase */
+  };
+
+  /**
+   * Resets the state if all file/s have been uploaded.
+   */
+  resetState = () => {
+    const { total, filesLength } = this.state;
+    if (total === filesLength) {
+      this.setState({ files: [], filesLength: 0, total: 0 });
+    }
+  };
+
+  /**
+   * Will read the file/s, check if multiple files can be uploaded,
+   * check for errors, if no errors then upload the file.
+   * @param {Object} files Selected files.
+   */
+  readFile = files => {
+    const {
+      setBorder,
+      checkFile,
+      uploadFile,
+      props: {
+        inputProps: { multiple },
+      },
+    } = this;
+
+    this.setState({ filesLength: files.length, errors: [] });
+    setBorder();
+
+    if (multiple) {
+      Object.keys(files).forEach(key => {
+        const file = files[key];
+        if (checkFile(file)) {
+          uploadFile(file);
+        }
+      });
+    }
+
+    // If single file upload, upload the first file from the dropped files
+    if (!multiple && checkFile(files[0])) {
+      uploadFile(files[0]);
+    }
+  };
+
+  render = () => {
+    const {
+      onDrop,
+      onClick,
+      getFiles,
+      onDragOver,
+      onDragLeave,
+      state: { errors },
+      props: {
+        inputProps: { multiple },
+      },
+    } = this;
+
     return (
-      <Element
-        drag={this.state.drag}
-        onDragOver={this.onDragOver}
-        onDragLeave={this.onDragLeave}
-        onDrop={this.onDrop}
-      >
-        {(this.state.file &&
-          this.state.file.filename &&
-          this.state.file.filename[0].value) ||
-          'Drop a file or click here to upload one.'}
-      </Element>
+      <Card>
+        <CardContent>
+          <Element
+            onDrop={onDrop}
+            onClick={onClick}
+            onDragOver={onDragOver}
+            onDragLeave={onDragLeave}
+            innerRef={element => {
+              this.element = element;
+            }}
+          >
+            <Text>
+              <p>
+                {multiple
+                  ? 'Drop files or click here to upload.'
+                  : 'Drop a file or click here to upload.'}
+              </p>
+              <Button
+                size="small"
+                color="primary"
+                variant="contained"
+                aria-label="Upload Image/s"
+              >
+                Upload <CloudUploadIcon className="icon" />
+              </Button>
+            </Text>
+
+            <input
+              type="file"
+              onChange={getFiles}
+              multiple={multiple}
+              style={{ display: 'none' }}
+              ref={element => {
+                this.input = element;
+              }}
+            />
+          </Element>
+
+          {errors.length > 0 && (
+            <Error>
+              <p>One or more files could not be uploaded.</p>
```

Comment:
Could we use some material UI components for displaying these error messages maybe?

Reply:
I have updated this to use the typography component that Material supplies.